public/app/plugins/datasource/prometheus/querybuilder/shared/operationUtils.ts

```diff
@@ -321,3 +322,45 @@ export function getOnLabelAddedHandler(changeToOperationId: string) {
     return op;
   };
 }
+
+export function isConflictingSelector(
+  newLabel: Partial<QueryBuilderLabelFilter>,
+  labels: Array<Partial<QueryBuilderLabelFilter>>
+): boolean {
+  if (labels.length < 2) {
+    return false;
+  }
+
+  if (!newLabel.label || !newLabel.op || !newLabel.value) {
+    return false;
+  }
+
+  let isConflicting = false;
+
+  const labelsWithoutNew = labels.filter((label) => {
+    if (label.label === newLabel.label && label.op === newLabel.op && label.value === newLabel.value) {
+      return;
+    }
+
+    if (!label.label || !label.op || !label.value) {
+      return;
+    }
+
+    return label;
+  });
+
+  labelsWithoutNew.forEach((label) => {
+    if (label.label !== newLabel.label || label.value !== newLabel.value) {
+      return;
+    }
+
+    if (
+      (label.op!.startsWith('!') && !String(newLabel.op).startsWith('!')) ||
```

Comment:
I think this could be simplified to just `label.op?.startsWith('!') && !newLabel.op?.startsWith('!')`